crates/hir-expand/src/fixup.rs

```diff
@@ -571,6 +559,17 @@ mod tests {
             parse.syntax_node()
         );
 
+        // the fixed-up tree should not contain braces as punct
+        // FIXME: should probably instead check that it's a valid punctuation character
+        for x in tt.iter() {
```

Comment:
Perhaps it's worth gating it under `cfg(debug_assertions)` for perf?

Reply:
Also note that this needs to iterator over `flat_tokens()` instead, as written it won't validate grandchildren (and also be a bit slower. Iterating over flat tokens is okay here since we only look for leafs).

Reply:
This is just in the test, so it should be fine. You're right about `flat_tokens`, oops.